### PR TITLE
 Intake withdraw issues

### PR DESCRIPTION
**Pull request by Sjones352**
connects #9986

Screen Shot 2019-04-15 at 4 42 55 PM

### CODE REVIEW

File: app/models/request_issue.rb [hunk 1]
```diff
@@ -143,6 +143,14 @@ def active_or_ineligible
       active.or(ineligible)
     end
 
+    def withdrawn
+      eligible.where(closed_status: :withdrawn)
+    end
+
+    def active_or_ineligible_withdrawn
```
Comment (pkarman) [inline app/models/request_issue.rb]: could we rename this `active_or_ineligible_or_withdrawn` for parity with the existing method?

File: app/models/request_issue.rb [hunk 2]
```diff
@@ -255,6 +263,11 @@ def special_issues
     return specials unless specials.empty?
   end
 
+  def withdrawal_date
```
Comment (pkarman) [inline app/models/request_issue.rb]: I like this!

File: app/models/request_issue.rb [hunk 3]
```diff
@@ -255,6 +263,11 @@ def special_issues
...
+    closed_at if closed_status == "withdrawn"
```
Comment (pkarman) [inline app/models/request_issue.rb]: I think the enum of `closed_status` means Rails gives us the magical `withdrawn?` method so we don't have to check `closed_status == "withdrawn"`

File: spec/feature/intake/higher_level_review/edit_spec.rb [hunk 4]
```diff
@@ -1172,7 +1172,6 @@ def click_cancel(visit_page)
         expect(page).to have_content("Requested issues\n1. PTSD denied")
 
         click_withdraw_intake_issue_dropdown("PTSD denied")
-
```
Comment (pkarman) [inline spec/feature/intake/higher_level_review/edit_spec.rb]: I try and keep a blank line between the set up for the `expect` and the `expect` lines themselves. Obviously the code base is inconsistent about this as it is. I find the extra whitespace easier to read where the "actual test" is vs the previous setup steps.

Comment (leikkisa) [reply spec/feature/intake/higher_level_review/edit_spec.rb]: 👍

File: spec/feature/intake/supplemental_claim/edit_spec.rb [hunk 5]
```diff
@@ -638,6 +638,51 @@ def click_cancel(visit_page)
         expect(withdrawn_issue.decision_review.end_product_establishments.first.synced_status).to eq("CAN")
         expect(Fakes::VBMSService).to have_received(:remove_contention!).once
       end
+
+      fscenario "show withdrawn issue when edit page is reloaded" do
```
Comment (leikkisa) [inline spec/feature/intake/supplemental_claim/edit_spec.rb]: There is a focus here

File: spec/feature/intake/appeal/edit_spec.rb [hunk 6]
```diff
@@ -495,6 +495,53 @@
...
+      expect(withdrawn_issue.closed_at).to eq(1.day.ago.to_date.to_datetime)
+
+      # reload to verify that the new issues populate the form
+      visit "appeals/#{appeal.uuid}/edit/"
+
+      expect(page).to have_content("Requested issues\n1. Left knee granted")
+      expect(page).to have_content(
+        /Withdrawn issues\n[1-2]..PTSD denied\nDecision date: 01\/20\/2018\nWithdrawn on/i
+      )
+      # expect(page).to have_content("Withdrawn issues\n2. PTSD denied\nDecision date: 01/19/2018\nWithdrawn on 02/07/2018")
```
Comment (leikkisa) [inline spec/feature/intake/appeal/edit_spec.rb]: Should this be commented out?

File: spec/feature/intake/higher_level_review/edit_spec.rb [hunk 7]
```diff
@@ -1194,6 +1193,51 @@ def click_cancel(visit_page)
...
+        withdrawn_issue = RequestIssue.where(closed_status: "withdrawn").first
+        expect(withdrawn_issue).to_not be_nil
+        expect(withdrawn_issue.closed_at).to eq(1.day.ago.to_date.to_datetime)
+
+        # reload to verify that the new issues populate the form
+        visit "higher_level_reviews/#{rating_ep_claim_id}/edit"
+
+        expect(page).to have_content("Requested issues\n1. Left knee granted")
+        expect(page).to have_content("Withdrawn issues\n2. PTSD denied\nDecision date: 01/19/2018\nWithdrawn on 02/07/2018")
+        expect(withdrawn_issue.closed_at).to eq(1.day.ago.to_date.to_datetime)
```
Comment (pkarman) [inline spec/feature/intake/higher_level_review/edit_spec.rb]: why is the withdrawal date yesterday? why not right now?

is there a timezone offset bug here?

**Pull request merged**